extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -105,7 +105,14 @@ export class AmpIframe extends AMP.BaseElement {
         !((' ' + sandbox + ' ').match(/\s+allow-same-origin\s+/i)),
         'allow-same-origin is not allowed with the srcdoc attribute %s.',
         this.element);
-    return 'data:text/html;charset=utf-8;base64,' + btoa(srcdoc);
+
+    // srcdoc is a DOMString and requires special handling for base64 encoding.
+    // See https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding
+    const encodedSrcdoc = btoa(encodeURIComponent(srcdoc)
```

Comment:
I tried a few designs and decided mixing base64 encoding and string encoding does not make sense since they are completely different concepts. I will send a PR soon with the code and the reasoning behind the design.